src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnArrInit2.java

```diff
@@ -0,0 +1,53 @@
+/* Config:                                                                  //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration: //indent:1 exp:1
+ *                                                                          //indent:1 exp:1
+ * arrayInitIndent = 0                                                      //indent:1 exp:1
+ * basicOffset = 2                                                          //indent:1 exp:1
+ * braceAdjustment = 0                                                      //indent:1 exp:1
+ * caseIndent = 4                                                           //indent:1 exp:1
+ * forceStrictCondition = false                                             //indent:1 exp:1
+ * lineWrappingIndentation = 0                                              //indent:1 exp:1
+ * tabWidth = 8                                                             //indent:1 exp:1
+ * throwsIndent = 4                                                         //indent:1 exp:1
+ */                                                                         //indent:1 exp:1
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+@InputIndentationAnnArrInit2.Foo({ //indent:0 exp:0
+    @InputIndentationAnnArrInit2.Bar, //indent:4 exp:0,33,35 warn
```

Comment:
@romani `forceStrictCondition` is set to `false`. Shouldn't there be no violation here?

Reply:
@Abhishek-kumar09 , can you explain ?
This option allow to use extra spaces if that visually benefitial

Reply:
Actually it is not following forceStrictCondition.
If it would, then all the Indent-levels after arrayInitIndent/linewrappingIndent  is valid.
Particularly here, the user intent is to align code with 0 lineWrapping and 0 arrayInit indent.

For giving some flexibility over only `arrayInitIndent`, the child is also allowed to have `lineWrapping` and `lcurly+lineWrapping`.

But, yes it feels valid to me now that the child should also follow `forceStrictCondition` being false/true.

What's your take? Should I change the code?

Reply:
@rnveach I have applied the changes. Please have a look.

Reply:
sorry for being late on this discussion .... I needed some time to meditate on this.

https://checkstyle.org/config_misc.html#Indentation
from documentation

> forceStrictCondition | Force strict indent level in line wrapping case. If value is true, line wrap                   indent have to be same as lineWrappingIndentation parameter. If value is false,                   line wrap indent could be bigger on any value user would like.

it is clearly not for `arrayInitIndent`. arrayInitIndent is somehow in between of basicOffset and lineWrappingIndentation. Because technically array should be in one line and we just wrap array by `,` in multiple line.
But the same is true for basicOffset, Class/block can be single line `class A { int i; int j; }` and we just wrap it by `;` in separate lines. But we have special property for this - basicOffset. So wrapping by separator between completely independent items should follow some `offset`.

Lets come back to array.
Array in very similar to blocks, as all its inner elements are independent, but not separated by `;` so it has special property.
`lineWrappingIndentation` is more applicable for wrapping inside statement - it wrapping in level deeper than by `;`. For arrays - it is for wrapping inside array element, not between them.

@rnveach , @Abhishek-kumar09 if you both agree ...... Please revert your implementation of `lineWrappingIndentation` for arrays. Sorry for being late.

Reply:
@rnveach , @Abhishek-kumar09 , please read and if you agree lets proceed in this way.